src/api/search/src/search.js

```diff
@@ -76,4 +76,93 @@ const search = async (
   };
 };
 
-module.exports = search;
+/**
+ * Advanced search allows you to look up multiple or single fields based on the input provided
+ * @param postText    - text to search in post field
+ * @param authorText  - text to search in author field
+ * @param titleText   - text to search in title field
+ * @param dateStart   - published after this date
+ * @param dateEnd     - published before this date
+ * @return all the results matching the fields text
+ * Range queries: https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-query-string-query.html#_ranges
+ * Match field queries: https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-ppmatch-query.html#query-dsl-match-query-zero
+ */
+const advancedSearch = async (options) => {
+  const query = {
+    query: {
+      bool: {
+        must: [],
+      },
+    },
+  };
+
+  const must = query.query.bool.must;
+
+  if (options.author) {
+    must.push({
+      match: {
+        author: {
+          query: options.author,
```

Comment:
Above you say `authorText` and `postText` but in this code you use `options.author` and `options.post`.  Please confirm which one is right, and correct the other.

Reply:
I updated to the simpler one, but I need to update the comments you are correct.